main_test.go

```diff
@@ -224,6 +475,211 @@ func TestPublishersEndpoints(t *testing.T) {
 			expectedBody:        `{"title":"token authentication failed","status":401}`,
 			expectedContentType: "application/problem+json",
 		},
+		{
+			query: "POST /v1/publishers with invalid JSON",
+			body:  `INVALID_JSON`,
+			headers: map[string][]string{
+				"Authorization": {goodToken},
+				"Content-Type":  {"application/json"},
+			},
+			expectedCode:        400,
+			expectedContentType: "application/problem+json",
+			validateFunc: func(t *testing.T, response map[string]interface{}) {
+				assert.Equal(t, `can't create Publisher`, response["title"])
+				assert.Equal(t, "invalid json", response["detail"])
+			},
+		},
+		{
+			description: "POST publishers with validation errors",
+			query:       "POST /v1/publishers",
+			body:        `{"codeHosting": [{"url" : "a"}], "email":"b"}`,
+			headers: map[string][]string{
+				"Authorization": {goodToken},
+				"Content-Type":  {"application/json"},
+			},
+			expectedCode:        422,
+			expectedContentType: "application/problem+json",
+			validateFunc: func(t *testing.T, response map[string]interface{}) {
+				assert.Equal(t, `can't create Publisher`, response["title"])
+				assert.Equal(t, "invalid format", response["detail"])
+
+				assert.IsType(t, []interface{}{}, response["validationErrors"])
+
+				validationErrors := response["validationErrors"].([]interface{})
+				assert.Equal(t, 1, len(validationErrors))
+
+				firstValidationError := validationErrors[0].(map[string]interface{})
+
+				for key := range firstValidationError {
+					assert.Contains(t, []string{"field", "rule", "value"}, key)
+				}
+			},
+		},
+		{
+			description: "POST publishers with empty body",
+			query:       "POST /v1/publishers",
+			body:        "",
+			headers: map[string][]string{
+				"Authorization": {goodToken},
+				"Content-Type":  {"application/json"},
+			},
+			expectedCode:        400,
+			expectedContentType: "application/problem+json",
+			validateFunc: func(t *testing.T, response map[string]interface{}) {
+				assert.Equal(t, `can't create Publisher`, response["title"])
+				assert.Equal(t, "invalid json", response["detail"])
+			},
+		},
+		{
+			description: "PATCH non-existing publishers",
+			query:       "PATCH /v1/publishers/NO_SUCH_publishers",
+			body:        `{"codeHosting": [{"url" : "https://www.example.com"}], "email":"example@example.com"}`,
+			headers: map[string][]string{
+				"Authorization": {goodToken},
+				"Content-Type":  {"application/json"},
+			},
+			expectedCode:        404,
+			expectedBody:        `{"title":"Not Found","detail":"update publisher error: : can't update Publisher. Publisher was not found","status":404}`,
+			expectedContentType: "application/problem+json",
+		},
+		//TODO fix database locked test
```

Comment:
Have you tried explicitly loading fixtures for each test? 

```
fixtures:    []string{"publishers.yml", "publishers_code_hosting.yml"},
```

Reply:
Tried, but fails :-(